src/order/copy.lean

```diff
@@ -92,5 +92,5 @@ def conditionally_complete_lattice.copy (c : conditionally_complete_lattice α)
   conditionally_complete_lattice α :=
 begin
   refine { le := le, sup := sup, inf := inf, Sup := Sup, Inf := Inf, ..},
-  all_goals { subst_vars, unfreezeI, cases c, assumption }
+  all_goals { abstract { subst_vars, unfreezeI, cases c, assumption } }
```

Comment:
Let me quickly explain the cause of this timeout before I forget it.  With the new structure instances, terms contain the unreduced fields, e.g. `(≤)`.  In particular, these fields now contain a full copy of the structure as it is stated in the definition.  For example antisymmetry is stated in terms of a preorder:
```lean
∀ (a b : α),
    @has_le.le.{u} α
      (@preorder.to_has_le.{u} α
         (@preorder.mk.{u} α le (@lattice.lt._default.{u} α le)
            (@conditionally_complete_lattice.copy._aux_1.{u} α c le eq_le sup eq_sup inf eq_inf Sup eq_Sup Inf eq_Inf)
            (@conditionally_complete_lattice.copy._aux_2.{u} α c le eq_le sup eq_sup inf eq_inf Sup eq_Sup Inf eq_Inf)
            (λ (a b : α), iff.refl (@has_lt.lt.{u} α (@has_lt.mk.{u} α (@lattice.lt._default.{u} α le)) a b))))
      a
      b →
    @has_le.le.{u} α
      (@preorder.to_has_le.{u} α
         (@preorder.mk.{u} α le (@lattice.lt._default.{u} α le)
            (@conditionally_complete_lattice.copy._aux_1.{u} α c le eq_le sup eq_sup inf eq_inf Sup eq_Sup Inf eq_Inf)
            (@conditionally_complete_lattice.copy._aux_2.{u} α c le eq_le sup eq_sup inf eq_inf Sup eq_Sup Inf eq_Inf)
            (λ (a b : α), iff.refl (@has_lt.lt.{u} α (@has_lt.mk.{u} α (@lattice.lt._default.{u} α le)) a b))))
      b
      a →
    @eq.{u+1} α a b
```

The `abstract` tactic introduces these `*._aux*` definitions to abbreviate the generated proofs.  Otherwise they would be expanded and we'd need to substitute (`subst_vars`) the previous proofs in every goal, leading to horrible performance.